File: IPaddress/config/CODE/geolocate.py
import requests

class Data:
    def __init__(self, city, region, country, loc, org, postal):
        self.city = city
        self.region = region
        self.country = country
        self.loc = loc
        self.org = org
        self.postal = postal

def main():
    print("Geolocate")
    ip = input("Enter IP address: ")
    url = f"https://ipinfo.io/{ip}/json"

    try:
        response = requests.get(url)
        response.raise_for_status()

        print("[+] Request Successfully Made")
        ip_info = response.json()

        print(f"Country: {ip_info['country']}")
        print(f"City: {ip_info['city']}")
        print(f"Region: {ip_info['region']}")
        print(f"Coordinates: {ip_info['loc']}")
        print(f"ASN: {ip_info['org']}")
        print(f"Postal Code: {ip_info['postal']}")

        coords = ip_info['loc'].split(',')
        print(f"Google Maps: https://www.google.com/maps/?q={coords[0]},{coords[1]}")
    except requests.RequestException as ex:
        print(f"Error: {ex}")

if __name__ == "__main__":
    main()

File: CODE/geolocate.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace IPaddress
{
    public class Data
    {
        public string City { get; set; }
        public string Region { get; set; }
        public string Country { get; set; }
        public string Loc { get; set; }
        public string Org { get; set; }
        public string Postal { get; set; }
    }

    internal class Program
    {
        static async Task Main(string[] args)
        {
            Console.Title = "Geolocate";
            Console.Write("Enter IP address: ");
            string ip = Console.ReadLine();

            if (string.IsNullOrWhiteSpace(ip))
            {
                Console.WriteLine("Invalid IP address.");
                return;
            }

            try
            {
                Data ipInfo = await GetIpInfoAsync(ip);

                if (ipInfo != null)
                {
                    Console.Clear();
                    Console.WriteLine($"Country: {ipInfo.Country}");
                    Console.WriteLine($"City: {ipInfo.City}");
                    Console.WriteLine($"Region: {ipInfo.Region}");
                    Console.WriteLine($"Coordinates: {ipInfo.Loc}");
                    Console.WriteLine($"ASN: {ipInfo.Org}");
                    Console.WriteLine($"Postal Code: {ipInfo.Postal}");

                    string[] Coords = ipInfo.Loc.Split(',');
                    Console.WriteLine($"Google Maps: https://www.google.com/maps/?q={Coords[0]},{Coords[1]}");
                }
                else
                {
                    Console.WriteLine("Failed to retrieve IP information.");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"An error occurred: {ex.Message}");
            }
        }

        private static async Task<Data> GetIpInfoAsync(string ip)
        {
            string url = $"https://ipinfo.io/{ip}/json";

            using (HttpClient client = new HttpClient())
            {
                try
                {
                    HttpResponseMessage response = await client.GetAsync(url);
                    response.EnsureSuccessStatusCode();
                    Console.WriteLine("[+] Request Successfully Made");

                    string responseData = await response.Content.ReadAsStringAsync();
                    return JsonConvert.DeserializeObject<Data>(responseData);
                }
                catch (HttpRequestException httpRequestException)
                {
                    Console.WriteLine($"Request error: {httpRequestException.Message}");
                }
                catch (Exception ex)
                {
                    Console.WriteLine($"An error occurred: {ex.Message}");
                }

                return null;
            }
        }
    }
}